superset/dao/base.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Dict, Optional
+
+from flask_appbuilder.models.filters import BaseFilter
+from flask_appbuilder.models.sqla import Model
+from flask_appbuilder.models.sqla.interface import SQLAInterface
+from sqlalchemy.exc import SQLAlchemyError
+
+from superset.commands.exceptions import (
+    CreateFailedError,
+    DeleteFailedError,
+    UpdateFailedError,
+)
+from superset.extensions import db
+
+
+class BaseDAO:
+    model_cls: Optional[Model] = None
+    base_filter: Optional[BaseFilter] = None
+
+    @classmethod
+    def find_by_id(cls, model_id: int) -> Model:
```

Comment:
It's unclear what `cls` is here. Ideally, we would just have a sig like `def find_by_id(id: int) -> Model`